zgrab2_schemas/zgrab2/redis.py

```diff
@@ -25,6 +25,22 @@
         ]),
         "quit_response": String(doc="The response to the QUIT command.", examples=["OK"]),
         "version": String(doc="The version string, read from the the info_response (if available)."),
```

Comment:
It may be helpful to also have `version` broken out into major/minor/micro ints (assuming the above `redis_version:4.0.7` example is representative) -- `String` should let you do wildcard searching, it can be extremely painful to do int range searches using string operations.

Not a blocker, since you can always new fields later, and I think in any case we want to keep `version` as a `String` (e.g. for CVE matching).

Reply:
Just to clarify, I should have both `redis_version` as a string, and `redis_major`, `redis_minor`, and `redis_micro` as ints?

Reply:
Yeah, something like that -- probably matching whatever nomenclature they use for the segments (major/minor/revision or patch or whatever). The string will let you catch weird stuff like "1.04-beta" or "PRERELEASE-1.54.0" or "\r\nContent-Type: application/octet-stream", but the ints will help if you need to actually tract down `1.5.13 < version < 1.6.2`

Reply:
How would I go about differentiating between having a `0` within the version and not having the information? For example, given the version `5.0.0`, both the `minor` and `patchlevel` would not appear since `0` is the zero value of an int

Reply:
You could go the route of having the ints be pointers, so you can distinguish absent from 0, or just use 0 and add a note saying that if you do need to distinguish those, you need to look at the string; but my feeling would be that unless you are seeing a significant number of hosts with missing parts, the latter would be fine.

Reply:
from my 1%, it seems like >60% of the hosts have at least one section that's a 0

Reply:
That would be fine, as long as (say) < 1% of them are _missing_ a section.

I guess I should have clarified that in that case, you would not be using `omitempty`, since `0` is meaningful there.

Reply:
Of the redis servers that are found, most are hidden behind auth. So not using `omitempty` would mean that on those scenarios where we get a response but aren't able to gather any more information on them, we will be storing the major, minor, and patchlevel as 0's instead of omitting them. So I decided to take your initial idea of having the ints be pointers!